app/services/location_suggestion.rb

```diff
@@ -0,0 +1,50 @@
+class LocationSuggestion
+  MINIMUM_LOCATION_INPUT_LENGTH = 3
+  NUMBER_OF_SUGGESTIONS = 5
+
+  class InsufficientLocationInput < StandardError; end
+  class MissingLocationInput < StandardError; end
+  class GooglePlacesAutocompleteError < StandardError; end
+
+  attr_accessor :location_input
+
+  def initialize(location_input)
+    raise MissingLocationInput if location_input.nil?
+    raise InsufficientLocationInput if location_input.length < MINIMUM_LOCATION_INPUT_LENGTH
+    self.location_input = location_input
+  end
+
+  def suggest_locations
+    predictions = get_suggestions_from_google['predictions'].take(NUMBER_OF_SUGGESTIONS)
+    suggestions = predictions.map { |prediction| prediction['description'] }
+    matched_terms = predictions.map do
+      |prediction| prediction['terms'].select { |term| term['offset'] == 0 }.map { |hit| hit['value'] }
+    end
+    return suggestions, matched_terms
```

Comment:
no need to return on the last line of the method

Reply:
I thought when you want to return multiple values you do need to

Reply:
[suggestions, matched_terms]